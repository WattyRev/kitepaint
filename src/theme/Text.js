import styled, { css } from "styled-components";

/**
 * Displays styled text with the provided tag
 * @param {Boolean} isLight If set to true, the text will be lighter colored for dark backgrounds
 */
export const TypographyStyles = css`
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  line-height: 16px;
`;
const StyledText = styled.p`
  margin: 0;
  padding: 0;
  ${TypographyStyles}

  color: ${props =>
    props.isLight ? props.theme.colors.silver : props.theme.colors.black};
`;
export default StyledText;
