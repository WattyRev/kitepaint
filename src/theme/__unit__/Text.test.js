import React from "react";
import { mount, shallow } from "enzyme";
import Theme from "../../theme";
import Text from "../Text";

describe("Text", () => {
  it("renders", () => {
    expect.assertions(1);
    const wrapper = shallow(<Text theme={Theme}>button</Text>);
    expect(wrapper).toMatchSnapshot();
  });
  describe("isLight", () => {
    it("renders with light text if isLight is true", () => {
      const wrapper = mount(
        <Text theme={Theme} isLight>
          button
        </Text>
      );
      expect(wrapper).toHaveStyleRule("color", Theme.colors.silver);
      expect(wrapper).not.toHaveStyleRule("color", Theme.colors.black);
    });
    it("renders with dark text if isLight is false", () => {
      const wrapper = mount(
        <Text theme={Theme} isLight={false}>
          button
        </Text>
      );
      expect(wrapper).not.toHaveStyleRule("color", Theme.colors.silver);
      expect(wrapper).toHaveStyleRule("color", Theme.colors.black);
    });
  });
});
