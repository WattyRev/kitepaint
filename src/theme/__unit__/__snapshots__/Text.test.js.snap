// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef
    theme={
      Object {
        "colors": Object {
          "aqua": "#7FDBFF",
          "black": "#111111",
          "blue": "#0074D9",
          "blueDarker": "#005daa",
          "blueLighter": "#f0fbff",
          "fuchsia": "#F012BE",
          "gray": "#AAAAAA",
          "grayDark": "#5e5e5e",
          "green": "#2ECC40",
          "lime": "#01FF70",
          "maroon": "#85144B",
          "navy": "#001F3F",
          "olive": "#3D9970",
          "orange": "#FF851B",
          "purple": "#B10DC9",
          "red": "#FF4136",
          "silver": "#DDDDDD",
          "teal": "#39CCCC",
          "white": "#FFFFFF",
          "yellow": "#FFDC00",
        },
      }
    }
  >
    button
  </ForwardRef>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "button",
      "forwardedClass": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "rules": Array [
            "
  margin: 0;
  padding: 0;
  ",
            "
  font-family: -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\",
    \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
",
            "

  color: ",
            [Function],
            ";
",
          ],
        },
        "displayName": "styled.p",
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "p",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "forwardedRef": null,
      "theme": Object {
        "colors": Object {
          "aqua": "#7FDBFF",
          "black": "#111111",
          "blue": "#0074D9",
          "blueDarker": "#005daa",
          "blueLighter": "#f0fbff",
          "fuchsia": "#F012BE",
          "gray": "#AAAAAA",
          "grayDark": "#5e5e5e",
          "green": "#2ECC40",
          "lime": "#01FF70",
          "maroon": "#85144B",
          "navy": "#001F3F",
          "olive": "#3D9970",
          "orange": "#FF851B",
          "purple": "#B10DC9",
          "red": "#FF4136",
          "silver": "#DDDDDD",
          "teal": "#39CCCC",
          "white": "#FFFFFF",
          "yellow": "#FFDC00",
        },
      },
    },
    "ref": null,
    "rendered": "button",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "button",
        "forwardedClass": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  margin: 0;
  padding: 0;
  ",
              "
  font-family: -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\",
    \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
",
              "

  color: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.p",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "p",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "forwardedRef": null,
        "theme": Object {
          "colors": Object {
            "aqua": "#7FDBFF",
            "black": "#111111",
            "blue": "#0074D9",
            "blueDarker": "#005daa",
            "blueLighter": "#f0fbff",
            "fuchsia": "#F012BE",
            "gray": "#AAAAAA",
            "grayDark": "#5e5e5e",
            "green": "#2ECC40",
            "lime": "#01FF70",
            "maroon": "#85144B",
            "navy": "#001F3F",
            "olive": "#3D9970",
            "orange": "#FF851B",
            "purple": "#B10DC9",
            "red": "#FF4136",
            "silver": "#DDDDDD",
            "teal": "#39CCCC",
            "white": "#FFFFFF",
            "yellow": "#FFDC00",
          },
        },
      },
      "ref": null,
      "rendered": "button",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
